R/Renderer.R

```diff
@@ -33,20 +33,42 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
-      parsed_yaml <- yaml_header
+      report_type <- reverse_yaml_field(yaml_header, "output")
       parsed_global_knitr <- sprintf(
-        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n```\n",
-        capture.output(dput(global_knitr))
+        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n%s\n```\n",
+        capture.output(dput(global_knitr)),
+        if (identical(report_type, "powerpoint_presentation")) {
+          paste(
+            "# additional function to parse code block into a table for PPT",
+            "library('flextable')",
+            "code_block <- function(code_block) {",
+            "  df <- data.frame(code_block)",
+            "  ft <- flextable::flextable(df)",
+            "  ft <- flextable::delete_part(ft, part = 'header')",
+            "  ft <- flextable::autofit(ft, add_h = 0)",
+            "  ft <- flextable::fontsize(ft, size = 7, part = 'body')",
+            "  ft <- flextable::bg(x = ft,bg = 'lightgrey')",
+            "  ft <- flextable::border_outer(ft)",
+            "  if(flextable_dim(ft)$widths > 8) {",
+            "    ft <- width(ft, width = 8)",
+            "  }",
+            "  ft",
+            "}",
+            collapse = "\n"
+          )
+        } else {
+          ""
+        }
```

Comment:
This is far too big to be an `if` within a call, assign it as a variable and pass that to `sprintf`.

Reply:
I kept this here as it was used once, but I think it would be better to turn it into a utility function. This will allow us to run unit tests as well and. Thanks for suggestion - I made the necessary updates.